File: in_Java/0082 Remove Duplicates from Sorted List II.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if(head == null) return head;
        ListNode d = new ListNode(0);
        d.next = head;
        ListNode pre = d;
        ListNode cur = head;
        ListNode next = cur.next;
        while(next != null){
            if(cur.val == next.val){
                while(next != null && cur.val == next.val){
                    next = next.next;
                }
                pre.next = next;
                cur = next;
                if(cur != null) next = cur.next;
            }else{
                pre = cur;
                cur = next;
                next = cur.next;
            }
        }
        return d.next;
    }
}

File: in_Python/0082 Remove Duplicates from Sorted List II.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if head == None:
            return head
        p = ListNode(0)
        pp = p
        p.next = head
        cur = head
        nex = head.next
        if nex == None:
            return head
        while nex != None:
            nex = cur.next
            f = 0
            while nex != None and nex.val == cur.val:
                f = 1
                nex = nex.next
            if f == 0:
                p = cur
                cur = nex
            else:
                cur = nex                
                p.next = cur
        return pp.next
        
        